File: src/main/java/no/fintlabs/access/control/repository/PackageAccessEntityRepository.java
package no.fintlabs.access.control.repository;

import jakarta.validation.constraints.NotNull;
import no.fintlabs.access.control.model.entity.ClientEntity;
import no.fintlabs.access.control.model.entity.PackageAccessEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Collection;

public interface PackageAccessEntityRepository extends JpaRepository<PackageAccessEntity, Integer> {

    Collection<PackageAccessEntity> findByClient(@NotNull ClientEntity client);
}

File: src/main/java/no/fintlabs/access/control/repository/PackageAccessEntityRepository.kt
package no.fintlabs.access.control.repository

import jakarta.validation.constraints.NotNull
import no.fintlabs.access.control.model.entity.ClientEntity
import no.fintlabs.access.control.model.entity.PackageAccessEntity
import org.springframework.data.jpa.repository.JpaRepository

interface PackageAccessEntityRepository : JpaRepository<PackageAccessEntity, Int> {
    fun findByClient(client: ClientEntity): Collection<PackageAccessEntity>
}
